### PR TITLE
feat: 3weu4xd - BSC beefy refactor [SC]

### PR DESCRIPTION
**Pull request by naruto0913**
Main updates.
1. Beefy adapter contract refactor
2. Update integration testing for beefy adapter
3. Minor updates in BSC library contract

### CODE REVIEW

**Review by daiki-iu — APPROVED** [review]
@naruto0913 LGTM!

File: v1-contract/contracts/libraries/HedgepieLibraryBsc.sol [hunk 1]
```diff
@@ -192,7 +192,7 @@ library HedgepieLibraryBsc {
                 amountToken,
                 _adapter.addr,
                 tokenAddr,
-                IAdapterBsc(_adapter.addr).swapRouter(),
```
Comment (0xmainnov) [inline v1-contract/contracts/libraries/HedgepieLibraryBsc.sol:195]: @naruto0913 
swapRouter is existed to swapping single tokens. And we need to separate it with _router which is used for LP token. We need to leave swapRouter here.

File: v1-contract/contracts/libraries/HedgepieLibraryBsc.sol [hunk 2]
```diff
@@ -211,14 +211,14 @@ library HedgepieLibraryBsc {
                 amountA,
                 _adapter.addr,
                 tokens[0],
-                IAdapterBsc(_adapter.addr).swapRouter(),
```
Comment (0xmainnov) [inline v1-contract/contracts/libraries/HedgepieLibraryBsc.sol:214]: @naruto0913 
swapRouter is existed to swapping single tokens. And we need to separate it with _router which is used for LP token. We need to leave swapRouter here.

File: v1-contract/contracts/libraries/HedgepieLibraryBsc.sol [hunk 3]
```diff
@@ -211,14 +211,14 @@ library HedgepieLibraryBsc {
...
+                _router,
                 wbnb
             );
             amountOut += swapforBnb(
                 amountB,
                 _adapter.addr,
                 tokens[1],
-                IAdapterBsc(_adapter.addr).swapRouter(),
```
Comment (0xmainnov) [inline v1-contract/contracts/libraries/HedgepieLibraryBsc.sol:221]: @naruto0913 
swapRouter is existed to swapping single tokens. And we need to separate it with _router which is used for LP token. We need to leave swapRouter here.

**Review by 0xmainnov — APPROVED** [review]

**Pull request merged**